pkg/archive/diff.go

```diff
@@ -40,12 +42,30 @@ func UnpackLayer(dest string, layer ArchiveReader) (size int64, err error) {
 		// Normalize name, for safety and for a simple is-root check
 		hdr.Name = filepath.Clean(hdr.Name)
 
-		if !strings.HasSuffix(hdr.Name, "/") {
+		// Windows does not support filenames with colons in them. Ignore
+		// these files. This is not a problem though (although it might
+		// appear that it is). Let's suppose a client is running docker pull.
+		// The daemon it points to is Windows. Would it make sense for the
+		// client to be doing a docker pull Ubuntu for example (which has files
+		// with colons in the name under /usr/share/man/man3)? No, absolutely
+		// not as it would really only make sense that they were pulling a
+		// Windows image. However, for development, it is necessary to be able
+		// to pull Linux images which are in the repository.
```

Comment:
Even once the registry knows the type of image, I think we'll need code in here to deal with error cases.  Meaning, someone managed to upload a windows tar with a ":" in a filename. And in those cases we need to decide what to do - I'd prefer to error in those cases.  Now, we can wait until later if you want, but then let's add a TODO type of comment as a reminder.

Also, have you considered uploading a Windows tar/image for testing purposes? Then we could make this code more like how it should be.